shared/actions/profile/index.js

```diff
@@ -67,8 +68,20 @@ function onUserClick(username: string): Constants.OnUserClick {
 
 function* _onUserClick(action: Constants.OnUserClick): SagaGenerator<any, any> {
   const {username} = action.payload
-  yield put(switchTo([profileTab]))
-  yield put(navigateAppend([{props: {username}, selected: 'profile'}], [profileTab]))
+  if (!username.includes('@')) {
+    yield put(switchTo([profileTab]))
+    yield put(navigateAppend([{props: {username}, selected: 'profile'}], [profileTab]))
+  } else {
+    const searchResult = yield select(Selectors.searchResultSelector, username)
+    const fullname = searchResult.rightFullname
```

Comment:
Pulling info out of the entities store is unsatisfying here, but the _onUserClick() only has access to the username clicked on rather than the other details we need on the new NonUserProfile component..